### PR TITLE
196 backward euler fortran

### PR DESCRIPTION
**Pull request by dwfncar**
Added fortran API for Backward Euler solvers.
Made time_step and test_accuracy as inputs to test_multiple_grid_cells in test_micm_api.F90.

### CODE REVIEW

**Review by boulderdaze — APPROVED** [review]
Looks good!

File: fortran/test/fetch_content_integration/test_micm_api.F90 [hunk 1]
```diff
@@ -271,12 +272,19 @@ subroutine test_multiple_grid_cells(micm, NUM_GRID_CELLS)
...
-      ASSERT_NEAR(concentrations((i_cell-1)*NUM_SPECIES+D_index), D, 5.0e-3)
-      ASSERT_NEAR(concentrations((i_cell-1)*NUM_SPECIES+E_index), E, 5.0e-3)
-      ASSERT_NEAR(concentrations((i_cell-1)*NUM_SPECIES+F_index), F, 5.0e-3)
+      print *, 'A', concentrations((i_cell-1)*NUM_SPECIES+A_index), A
+      print *, 'B', concentrations((i_cell-1)*NUM_SPECIES+B_index), B
+      print *, 'C', concentrations((i_cell-1)*NUM_SPECIES+C_index), C
+      print *, 'D', concentrations((i_cell-1)*NUM_SPECIES+D_index), D
+      print *, 'E', concentrations((i_cell-1)*NUM_SPECIES+E_index), E
+      print *, 'F', concentrations((i_cell-1)*NUM_SPECIES+F_index), F
+      print *
```
Comment (mattldawson) [inline fortran/test/fetch_content_integration/test_micm_api.F90:281]: please remove diagnostic output

File: fortran/test/fetch_content_integration/test_micm_api.F90 [hunk 2]
```diff
@@ -303,14 +315,58 @@ subroutine test_multiple_grid_cell_standard_Rosenbrock()
...
 
+    real(c_double), parameter :: time_step = 200
+    real, parameter           :: test_accuracy = 5.0e-3
+
     num_grid_cells = 3
     micm => micm_t( "configs/analytical", RosenbrockStandardOrder, num_grid_cells, error )
     ASSERT( error%is_success() )
 
-    call test_multiple_grid_cells( micm, num_grid_cells )
+    print *, 'test_multiple_grid_cells standard Rosenbrock'
```
Comment (mattldawson) [inline fortran/test/fetch_content_integration/test_micm_api.F90:325]: please remove diagnostic output

**Review by mattldawson — CHANGES_REQUESTED** [review]
looks great! just a couple minor suggestions

**Review by K20shores — APPROVED** [review]

**Review by mattldawson — APPROVED** [review]
looks great!

**Pull request merged**